docs/docs/adrs/adr-017-allowing-inactive-validators.md

```diff
@@ -0,0 +1,110 @@
+
+## Changelog
+* 15th May 2024: Initial draft
+
+
+## Status
+
+Proposed
+
+## Context
+
+Currently, only validators in the active set on the provider can validate on consumer chains, which limits the number of validators that can participate in Interchain Security (ICS). 
+Validators outside of the active set might be willing
+to validate on consumer chains, but we might not want to make the provider validator set larger, e.g. to not put more strain on the consensus engine.
+This runs the risk of leaving consumer chains with too few validators.
+
+The purpose of this ADR is to allow validators that are *not* part of the consensus process on the provider chain (because they are inactive)
+to validate on consumer chains.
+
+In the context of this ADR, "consensus validator set" is the set of validators participating in the consensus protocol, and "staking validator set" is the set of validators viewed as active by the staking module.
+
+Currently, the staking module, provider module, and CometBFT interact in this way:
+
+![inactivevals_before.png](../../figures/inactivevals_before.png)
+
+The staking module keeps a list of validators. The `MaxValidators` validators with the largest amount of stake are "active" validators. `MaxValidators` is a parameter of the staking module. The staking module sends these validators to CometBFT to inform which validators make up the next consensus validators, that is, the set of validators participating in the consensus process. Separately, the provider module reads the list of bonded validators and sends this to the consumer chain, after shaping it according to which validators are opted in and the parameters set by the consumer chain for allowlist, denylist, etc.
+
+## Decision
+
+The proposed solution to allow validators that are not participating in the consensus process on the provider (inactive validators) is to change 3 main things:
+
+a) increase the `MaxValidators` parameter of the staking module
+
+b) do *not* take the updates for CometBFT directly from the bonded validators in the staking module, by wrapping the staking modules `EndBlocker` with a dummy EndBlocker that doesn't return any validator updates. Instead, we adjust the provider module to return validator updates on its EndBlocker. These validator updates are obtained by *filtering* the bonded validators to send only the first `MaxProviderConsensusValidators` (sorted by largest amount of stake first) many validators to CometBFT
+
+c) use the enlarged list of bonded validators from the staking module as basis for the validator set that the provider module sends to consumer chains (again after applying power shaping and filtering out validatiors that are not opted in).
+
+In consequence, the provider chain can keep a reasonably-sized consensus validator set, while giving consumer chains a much larger pool of potential validators.
+
+![inactivevals_after.png](../../figures/inactivevals_after.png)
+
+
```

Comment:
What extra queries will be added? Also, are there any changes in the state?

Reply:
I think changes to the state are useful to list. I am not convinced queries are useful to list.
Are queries really part of architecture decisions that we should document? It feels like the wrong level of detail to me, but lmk if you disagree, I can add it

Reply:
I agree. Unless the queries are essential to the design, they shouldn't be in the ADR but in the docs.